### PR TITLE
jq_fuzz_execute: fix memory leak

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Fixes: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=64626

### CODE REVIEW

File: tests/jq_fuzz_execute.cpp [hunk 1]
```diff
@@ -28,8 +28,11 @@ extern "C" int LLVMFuzzerTestOneInput(uint8_t *data, size_t size) {
         }
         jv_free(next);
       }
-
-      // Do not free "input" as this is handled by jq_teardown.
+      else {
+        // Only free if input as invalid as otherwise jq_teardown
+        // frees it.
```
Comment (wader) [inline tests/jq_fuzz_execute.cpp:32]: So if understand it correctly: `jq_parse` will always return something allocated. If valid the input argument will end up on the stack by `jq_start` and freed during teardown. If invalid it needs to be freed explicitly?

File: tests/jq_fuzz_execute.cpp [hunk 2]
```diff
@@ -28,8 +28,11 @@ extern "C" int LLVMFuzzerTestOneInput(uint8_t *data, size_t size) {
         }
         jv_free(next);
       }
-
-      // Do not free "input" as this is handled by jq_teardown.
+      else {
```
Comment (wader) [inline tests/jq_fuzz_execute.cpp]: nit pick: `} else {` to follow style in other files?

File: tests/jq_fuzz_execute.cpp (hunk 1 above)
Comment (DavidKorczynski) [reply tests/jq_fuzz_execute.cpp:32]: Yes, that's how I understand it.

File: tests/jq_fuzz_execute.cpp (hunk 2 above)
Comment (DavidKorczynski) [reply tests/jq_fuzz_execute.cpp]: Ah yes, thanks! Done!

**Review by wader — APPROVED** [review]
LGTM but maybe wait for one more approve

File: tests/jq_fuzz_execute.cpp (hunk 1 above)
Comment (emanuele6) [reply tests/jq_fuzz_execute.cpp:32]: "is invalid"?

Comment (DavidKorczynski) [reply tests/jq_fuzz_execute.cpp:32]: > "is invalid"?

Ah yes, good catch. Fixed.

**Comment by emanuele6** [comment]
Thank you!

**Pull request merged**